pkg/registry/apis/provisioning/resources/dualwriter.go

```diff
@@ -264,7 +269,8 @@ func (r *DualReadWriter) createOrUpdate(ctx context.Context, create bool, opts D
 	// Behaves the same running sync after writing
 	// FIXME: to make sure if behaves in the same way as in sync, we should
 	// we should refactor the code to use the same function.
-	if opts.Ref == "" && parsed.Client != nil {
+	shouldUpdate := opts.Ref == "" || (opts.Ref == opts.Branch && opts.Branch != "")
```

Comment:
In the condition after ||, opts.branch not being empty is implied.

Is opts.branch being empty just invalid state? I wonder if checking it separately above would be better for readability.

Reply:
Yeap refactored it, was WIP for testing made a util to be used across the places that made this check